### PR TITLE
5618 Create an automation test where the Tab Epidemiological data fro…

### PR DESCRIPTION
**Pull request by georgeginguta**
EpidemiologicalDataCase
…m Case is covered

Fixes #

### CODE REVIEW

**Comment by rdutu-vg** [comment]
Please add a proof of execution for those 15 runs.
Also, there are unused variables, which are also not filled in the test and compared, please cover those steps as well.

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/EpidemiologicalDataSteps.java [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+    webDriverHelpers.selectFromCombobox(
+        EpidemiologicalDataCasePage.COUNTRY_COMBOBOX, exposureData.getCountry());
+    //    webDriverHelpers.selectFromCombobox(
+    //        EpidemiologicalDataCasePage.STREET_INPUT, exposureData.getStreet());
+    //    webDriverHelpers.selectFromCombobox(
+    //        EpidemiologicalDataCasePage.HOUSE_NUMBER_INPUT, exposureData.getHouseNumber());
+    //    webDriverHelpers.selectFromCombobox(
+    //        EpidemiologicalDataCasePage.POSTAL_CODE_INPUT, exposureData.getPostalCode());
+    //    webDriverHelpers.selectFromCombobox(
+    //        EpidemiologicalDataCasePage.CITY_INPUT, exposureData.getCity());
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/EpidemiologicalDataSteps.java:185]: remove commented lines

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/EpidemiologicalDataSteps.java [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+            EpidemiologicalDataCasePage.LARGE_OUTBREAKS_AREA);
+    return YesNoUnknownOptions.valueOf(Text);
+  }
+
+  public YesNoUnknownOptions getContactsWithSource() {
+
+    String Text =
+        webDriverHelpers.getCheckedOptionFromHorizontalOptionGroup(
+            EpidemiologicalDataCasePage.CONTACTS_WITH_SOURCE_CASE_KNOWN);
+    return YesNoUnknownOptions.valueOf(Text);
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/EpidemiologicalDataSteps.java:277]: please remove all the above get methods and use them straight in the code where they are called

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/EpidemiologicalDataSteps.java [hunk 3]
```diff
@@ -0,0 +1,517 @@
...
+    String Text =
+        webDriverHelpers.getValueFromCombobox(EpidemiologicalDataCasePage.SUBCONTINENT_COMBOBOX);
+    return Text;
+  }
+
+  public String getCountry() {
+    String Text =
+        webDriverHelpers.getValueFromCombobox(EpidemiologicalDataCasePage.COUNTRY_COMBOBOX);
+    return Text;
+  }
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/EpidemiologicalDataSteps.java:516]: same as above, remove all get methods and put the lines straight where they are used

File: sormas-e2e-tests/src/main/java/org/sormas/e2etests/enums/cases/epidemiologicalData/TypeOfActivityActivity.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package org.sormas.e2etests.enums.cases.epidemiologicalData;
+
+public enum TypeOfActivityActivity {
```
Comment (alinapopaiustina) [inline sormas-e2e-tests/src/main/java/org/sormas/e2etests/enums/cases/epidemiologicalData/TypeOfActivityActivity.java:3]: Can we find another name that will not duplicate Activity?

**Review by alinapopaiustina — APPROVED** [review]

File: sormas-e2e-tests/src/main/java/org/sormas/e2etests/enums/cases/epidemiologicalData/TypeOfActivityActivity.java (hunk 4 above)
Comment (rdutu-vg) [reply sormas-e2e-tests/src/main/java/org/sormas/e2etests/enums/cases/epidemiologicalData/TypeOfActivityActivity.java:3]: if is the same, we can use the existent one

File: sormas-e2e-tests/src/main/java/org/sormas/e2etests/enums/cases/epidemiologicalData/ExposureDetailsRole.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    return role;
+  }
+
+  public static ExposureDetailsRole fromString(String role) {
+    for (ExposureDetailsRole b : ExposureDetailsRole.values()) {
+      if (b.role.equalsIgnoreCase(role)) {
+        return b;
+      }
+    }
+    return null;
```
[the remark this answers is not in the document]
Comment (rdutu-vg) [reply sormas-e2e-tests/src/main/java/org/sormas/e2etests/enums/cases/epidemiologicalData/ExposureDetailsRole.java:34]: change system out with log

File: sormas-e2e-tests/src/main/java/org/sormas/e2etests/enums/cases/epidemiologicalData/ActivityAsCaseType.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    return activityCase;
+  }
+
+  public static ActivityAsCaseType fromString(String activity) {
+    for (ActivityAsCaseType activityAsCaseType : ActivityAsCaseType.values()) {
+      if (activityAsCaseType.activityCase.equalsIgnoreCase(activity)) {
+        return activityAsCaseType;
+      }
+    }
+    System.out.println("Could not map " + activity + " to ActivityAsCaseType");
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/main/java/org/sormas/e2etests/enums/cases/epidemiologicalData/ActivityAsCaseType.java:26]: change system out with log

File: sormas-e2e-tests/src/main/java/org/sormas/e2etests/enums/cases/epidemiologicalData/TypeOfActivityExposure.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    return activity;
+  }
+
+  public static TypeOfActivityExposure fromString(String activity) {
+    for (TypeOfActivityExposure typeOfActivityExposure : TypeOfActivityExposure.values()) {
+      if (typeOfActivityExposure.activity.equalsIgnoreCase(activity)) {
+        return typeOfActivityExposure;
+      }
+    }
+    System.out.println("Could not map " + activity + " to TypeOfActivityExposure");
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/main/java/org/sormas/e2etests/enums/cases/epidemiologicalData/TypeOfActivityExposure.java:31]: change system out with log

File: sormas-e2e-tests/src/main/java/org/sormas/e2etests/enums/cases/epidemiologicalData/TypeOfPlace.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    return place;
+  }
+
+  public static TypeOfPlace fromString(String place) {
+    for (TypeOfPlace typeOfPlace : TypeOfPlace.values()) {
+      if (typeOfPlace.place.equalsIgnoreCase(place)) {
+        return typeOfPlace;
+      }
+    }
+    System.out.println("Could not map " + place + " to TypeOfPlace");
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/main/java/org/sormas/e2etests/enums/cases/epidemiologicalData/TypeOfPlace.java:29]: change system out with log

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/EpidemiologicalDataSteps.java [hunk 9]
```diff
@@ -0,0 +1,296 @@
...
+    webDriverHelpers.selectFromCombobox(
+        COUNTRY_COMBOBOX, exposureData.getCountry());
+    //    webDriverHelpers.selectFromCombobox(
+    //        STREET_INPUT, exposureData.getStreet());
+    //    webDriverHelpers.selectFromCombobox(
+    //        HOUSE_NUMBER_INPUT, exposureData.getHouseNumber());
+    //    webDriverHelpers.selectFromCombobox(
+    //        POSTAL_CODE_INPUT, exposureData.getPostalCode());
+    //    webDriverHelpers.selectFromCombobox(
+    //        CITY_INPUT, exposureData.getCity());
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/EpidemiologicalDataSteps.java:182]: remove commented lines

**Pull request merged**